src/pragha-playlist.c

```diff
@@ -3611,237 +3632,54 @@ create_current_playlist_columns(PraghaPlaylist *cplaylist, GtkTreeView *view)
-	gtk_widget_show(label_comment);
-	col_button = gtk_widget_get_ancestor(label_comment, GTK_TYPE_BUTTON);
-	g_signal_connect(G_OBJECT(GTK_WIDGET(col_button)), "button-press-event",
-			 G_CALLBACK(header_right_click_cb), cplaylist);
+    create_playlist_column(cplaylist, GTK_TREE_VIEW(view),P_TRACK_NO_STR,P_TRACK_NO);
+    create_playlist_column(cplaylist, GTK_TREE_VIEW(view),P_TITLE_STR,P_TITLE);
+    create_playlist_column(cplaylist, GTK_TREE_VIEW(view),P_ARTIST_STR,P_ARTIST);
+    create_playlist_column(cplaylist, GTK_TREE_VIEW(view),P_ALBUM_STR,P_ALBUM);
+    create_playlist_column(cplaylist, GTK_TREE_VIEW(view),P_GENRE_STR,P_GENRE);
+    create_playlist_column(cplaylist, GTK_TREE_VIEW(view),P_BITRATE_STR,P_BITRATE);
+    create_playlist_column(cplaylist, GTK_TREE_VIEW(view),P_YEAR_STR,P_YEAR);
+    create_playlist_column(cplaylist, GTK_TREE_VIEW(view),P_COMMENT_STR,P_COMMENT);
+    create_playlist_column(cplaylist, GTK_TREE_VIEW(view),P_LENGTH_STR,P_LENGTH);
+    create_playlist_column(cplaylist, GTK_TREE_VIEW(view),P_FILENAME_STR,P_FILENAME);
+    create_playlist_column(cplaylist, GTK_TREE_VIEW(view),P_MIMETYPE_STR,P_MIMETYPE);
+    
+}
 
-	/* Column : Length */
 
+static void
+create_playlist_column(PraghaPlaylist *cplaylist, GtkTreeView *view,gchar *label,gint column_id)
+{
+	GtkWidget *label_wgt;
+	GtkCellRenderer *renderer;
+	GtkTreeViewColumn *column;
+	GtkWidget *col_button;
+	
+	label_wgt = gtk_label_new(label);
+	
 	renderer = gtk_cell_renderer_text_new();
-	gtk_cell_renderer_text_set_fixed_height_from_font(GTK_CELL_RENDERER_TEXT(renderer),1);
 	gtk_cell_renderer_set_fixed_size (renderer, 1, -1);
-	column = gtk_tree_view_column_new_with_attributes(P_LENGTH_STR,
-							  renderer,
-							  "text",
-							  P_LENGTH,
-							  NULL);
-	gtk_tree_view_column_set_sizing(column, GTK_TREE_VIEW_COLUMN_FIXED);
-	gtk_tree_view_column_set_resizable(column, TRUE);
-	gtk_tree_view_column_set_sort_column_id(column, P_LENGTH);
-	gtk_tree_view_append_column(GTK_TREE_VIEW(view), column);
-	gtk_tree_view_column_set_widget(column, label_length);
-	gtk_widget_show(label_length);
-	col_button = gtk_widget_get_ancestor(label_length, GTK_TYPE_BUTTON);
-	g_signal_connect(G_OBJECT(GTK_WIDGET(col_button)), "button-press-event",
-			 G_CALLBACK(header_right_click_cb), cplaylist);
-
-	/* Column : Filename */
-
-	renderer = gtk_cell_renderer_text_new();
 	gtk_cell_renderer_text_set_fixed_height_from_font(GTK_CELL_RENDERER_TEXT(renderer),1);
-	gtk_cell_renderer_set_fixed_size (renderer, 1, -1);
-	column = gtk_tree_view_column_new_with_attributes(P_FILENAME_STR,
+	column = gtk_tree_view_column_new_with_attributes(label,
 							  renderer,
 							  "text",
-							  P_FILENAME,
+							  column_id,
 							  NULL);
 	gtk_tree_view_column_set_sizing(column, GTK_TREE_VIEW_COLUMN_FIXED);
 	gtk_tree_view_column_set_resizable(column, TRUE);
-	gtk_tree_view_column_set_sort_column_id(column, P_FILENAME);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
```

Comment:
Is interesting to be able to sort the columns .. But I should be able to keep the order between restarts of pragha ... :wink: 
Do you dare to do it? :stuck_out_tongue_winking_eye:

Reply:
I will try later . It breaks the add and delete columns to.